src/components/main/codeView/CodeView.tsx

```diff
@@ -72,36 +72,38 @@ export default function CodeView(props: CodeViewProps) {
 
   const isFirst = useRef<boolean>(true);
 
-  const codeContent = useRef<string>("");
+  const [codeContent, setCodeContent] = useState<string>("");
   const previewDiv = useRef(null);
 
   // -------------------------------------------------------------- sync --------------------------------------------------------------
 
   // build node tree reference
   useEffect(() => {
     const validNodeTreeRef = getValidNodeTreeInstance();
-    validNodeTreeRef.current = JSON.parse(JSON.stringify(validNodeTree));
+    validNodeTreeRef.current = structuredClone(validNodeTree);
```

Comment:
Is this just an improvement to use native feature or does this results in resolving the issue?

Reply:
just an improvement to use native feature